### PR TITLE
chore: add saml signup test / improve signup e2e test

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Add SAML signup flow test
- Simplify some code in signup e2e test
- Add test for cookie consent checkbox

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/modules/signup-view.tsx [hunk 1]
```diff
@@ -206,7 +206,7 @@ export default function Signup({
     }
   }, [redirectUrl]);
 
-  const [COOKIE_CONSENT, setCOOKIE_CONSENT] = useState(false);
+  const [COOKIE_CONSENT, setCOOKIE_CONSENT] = useState(false); // No need to be checked for user to proceed
```
Comment (keithwillcode) [inline apps/web/modules/signup-view.tsx:209]: I think this is the first time I've ever set something named with this casing 😄

Comment (emrysal) [reply apps/web/modules/signup-view.tsx:209]: Oh heeeel no 😆

Comment (hbjORbj) [reply apps/web/modules/signup-view.tsx:209]: yeah lol, let me change it

File: apps/web/playwright/signup.e2e.ts [hunk 2]
```diff
@@ -27,6 +27,12 @@ test.describe("Signup Main Page Test", async () => {
     await page.locator('[data-testid="continue-with-google-button"]').click();
     await page.waitForURL("/auth/sso/google");
   });
+
+  test("Continue with SAML button must exist / work", async ({ page }) => {
+    await page.goto("/signup");
+    await expect(page.locator('[data-testid="continue-with-saml-button"]')).toBeVisible();
+    await page.locator('[data-testid="continue-with-saml-button"]').click();
+  });
```
Comment (sean-brydon) [inline apps/web/playwright/signup.e2e.ts:35]: We should skip this based on if SAML is enabled or not on the instance - this PR fails tests locally for me as saml is not enabled for me locally.

File: apps/web/playwright/signup.e2e.ts [hunk 3]
```diff
@@ -331,4 +341,76 @@ test.describe("Email Signup Flow Test", async () => {
...
+      await expect(submitButton).toBeEnabled();
+
+      // the cookie consent checkbox does not need to be checked for user to proceed
+      await checkbox.uncheck();
+      await expect(submitButton).toBeEnabled();
+    });
+  });
+});
+
+test.describe("SAML Signup Flow Test", async () => {
```
Comment (sean-brydon) [inline apps/web/playwright/signup.e2e.ts:371]: Same here lets skip if saml is disabled on the instance

File: apps/web/playwright/signup.e2e.ts [hunk 4]
```diff
@@ -331,4 +341,76 @@ test.describe("Email Signup Flow Test", async () => {
...
+  test("Should navigate user to another URL", async ({ page }) => {
+    const submitButton = page.locator('[data-testid="saml-submit-button"]');
+    await page.goto("/signup");
+
+    // Navigate to email form
+    await page.locator('[data-testid="continue-with-saml-button"]').click();
```
Comment (sean-brydon) [inline apps/web/playwright/signup.e2e.ts:377]: NIT: lets try use `getByTestId`
 https://playwright.dev/docs/locators#locate-by-test-id

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Nice one thanks for adding tests here - lets consider skiping the SAML tests if they are disabled on the instance

**Comment by hbjORbj** [comment]
@sean-brydon Thanks Sean! Addressed

**Comment by keithwillcode** [comment]
@hbjORbj the test  is being marked as flaky. We should fix before merging. 

cc @anikdhabal for any tips since you’ve being working on tests recently 

1 flaky
    [@***com/web] › apps/web/playwright/saml.e2e.ts:40:9 › SAML tests › SAML Signup Flow Test › Submit button should be disabled without email

**Comment by anikdhabal** [comment]
> cc @anikdhabal for any tips since you’ve being working on tests recently

Yes, this is flaky right now due to a race condition. As we can see in the newly added tests, the form is immediately filled after navigating to the signup page, without waiting for the page to fully load or render. After the page finishes loading, it resets the form.
To fix this, we need to wait for the page to load before filling out the form. In the old test, we wait for the text to be completely translated and visible on the page before filling out the form

**Comment by hbjORbj** [comment]
Wow, thanks @anikdhabal !

**Review by sean-brydon — APPROVED** [review]
Happy with this! Thanks for making them changes to the tests!

Ship it!

**Pull request merged**